gnucash/gnome/dialog-trans-assoc.c

```diff
@@ -139,37 +141,40 @@ assoc_dialog_sort (AssocDialog *assoc_dialog)
     gtk_tree_sortable_set_sort_column_id (sortable, URI, order);
 }
 
-static const gchar *
-convert_uri_relative_to_uri (AssocDialog *assoc_dialog, const gchar *uri)
+static gchar *
+convert_uri_to_filename (AssocDialog *assoc_dialog, const gchar *uri, gchar *scheme)
 {
-    const gchar *new_uri;
+    gchar *file_path = NULL;
 
-    if (assoc_dialog->valid_path_head && g_str_has_prefix (uri,"file:/") &&
-        !g_str_has_prefix (uri,"file://")) // path is relative
+    if (!scheme) // relative path
     {
-        const gchar *part = uri + strlen ("file:");
-        new_uri = g_strconcat (assoc_dialog->path_head, part, NULL);
+        if (assoc_dialog->path_head_set) // not default entry
+            file_path = gnc_file_path_absolute (gnc_uri_get_path (assoc_dialog->path_head), uri);
+        else
+            file_path = gnc_file_path_absolute (NULL, uri);
     }
-    else
-        new_uri = g_strdup (uri);
 
-    return new_uri;
+    if (g_strcmp0 (scheme, "file") == 0) // absolute path
```

Comment:
See remark on other file about more generic file compatible scheme testing.

Reply:
I have changed most of these to use gnc_is_file_scheme but on the one in gnc_split_reg I have left it as that code will end up at the GtkFileChooser which seems correct to me...

Reply:
That's reasonable.